cypress/integration/common/transactions_steps.js

```diff
@@ -0,0 +1,73 @@
+import { And, Then } from 'cypress-cucumber-preprocessor/steps'
+import TransactionEndpoints from '../../endpoints/transaction_endpoints'
+
+And('I add {int} {word} transactions to it', (numberToAdd, transactionType) => {
+  cy.fixture(`${transactionType}.presroc.transaction`).then((transaction) => {
+    transaction.customerReference = 'C000000001'
+    transaction.licenceNumber = 'LIC/00000/01'
+
+    cy.get('@billRun').then((billRun) => {
+      // Due to the async nature of Cypress a classic `for` or `while` loop will not work because it will result in a
+      // non-deterministic execution order. So, we generate an array from our param using `Array.from`. For example,
+      // 5 would become [1, 2, 3, 4, 5]. We then use Cypress `each()` function to give us an async iterator!
+      //
+      // > Cypress each() - Iterate through an array like structure (arrays or objects with a length property).
+      // - solution provided by https://stackoverflow.com/a/53487016/6117745
+      const genArr = Array.from({ length: numberToAdd }, (v) => v)
+      cy.wrap(genArr).each(() => {
+        TransactionEndpoints.create(billRun.id, transaction)
+      })
+    })
+  })
+})
+
+And('I add a {word} {word} transaction to it', (transactionType, ruleset) => {
+  const fixtureName = `${transactionType}.${ruleset}.transaction`
+  const uuid = require('uuid')
+
+  const clientID = uuid.v4()
+
+  cy.fixture(fixtureName).then((fixture) => {
+    fixture.clientId = clientID
+    cy.get('@billRun').then((billRun) => {
+      TransactionEndpoints.create(billRun.id, fixture, false).then((response) => {
+        cy.wrap(response).as('transaction')
+      })
+    })
+  })
+})
+
+And('I add an invalid {word} transaction to it', (ruleset) => {
+  const fixtureName = `standard.${ruleset}.transaction`
+
+  cy.log(`Sending invalid '${ruleset}' transaction request`)
+
+  cy.fixture(fixtureName).then((fixture) => {
+    fixture.periodStart = ' '
+    cy.get('@billRun').then((billRun) => {
+      TransactionEndpoints.create(billRun.id, fixture, false).then((response) => {
+        cy.wrap(response).as('transaction')
+      })
+    })
+  })
+})
+
+Then('I get a successful response', () => {
```

Comment:
The only thing to be mindful of is anything in `/common` is always loaded for every scenario. This means you might mistakenly clash with another step if the step description is very generic. If you somehow ram the word 'transaction' in there you should be fine

```suggestion
Then('I get a successful response from the POST /transactions endpoint', () => {
```

🤷